contents/scalability/scalable-development.md

```diff
@@ -33,9 +33,29 @@ There are a few common bottlenecks that can be avoided such as database querying
 
 There are many program profilers available that you can use to find the bottlenecks in your server. By finding out where the server spends most computational time in, you can potentially save huge amounts of time.
 
-Performance bottlenecks ([resource](https://www.apicasystem.com/blog/5-common-performance-bottlenecks/))
+Performance bottlenecks ([resource](https://www.apicasystem.com/blog/5-common-performance-bottlenecks))
 
-Non-Relational Database ([resource](http://www.jamesserra.com/archive/2015/08/relational-databases-vs-non-relational-databases/)) ([resource](https://www.pluralsight.com/blog/software-development/relational-non-relational-databases))
+Non-Relational Database ([resource](http://www.jamesserra.com/archive/2015/08/relational-databases-vs-non-relational-databases)) ([resource](https://www.pluralsight.com/blog/software-development/relational-non-relational-databases))
+
+### Caching
+When more and more data is fetched from the database, the response becomes slower. This is where caching comes in. A cache is a key-value store that resides between the application and the database. By retrieving data from the cache instead of the database in the back-end server, the response time reduces greatly.
+
+The next thing to consider is what to cache. The rule of thumb is to cache data that is frequently accessed and read-mostly, meaning that an entry is read more often than it is updated. With this, cache hits would be more often than cache misses and as a result, the speed increase would outweigh the cost of maintaining the cache.  
+
+Caching ([resource](http://www.lecloud.net/post/9246290032/scalability-for-dummies-part-3-cache))
+
+Scalability best practices ([resource](https://www.infoq.com/articles/ebay-scalability-best-practices))
+
+### Go Asynchronous 
+Unlike synchronous operations, which runs sequentially and waits for the previous operation to complete before moving on to the next, asynchronous operations don't wait for other operations to finish before proceeding. Instead, they continue and allow other operations to finish.
+
+By using asynchronous operations, components are less tightly coupled and can be scaled independently of one another. As a result, if one component is down, the other component can still move forward.
+
+In situations in which responding to request is crucial, asynchronous operations can reduce the latency experienced by the requester. This is done by prioritising the quickness of the response to the user over the speed in which other processes complete.
```

Comment:
I find this segment kinda unclear too ><

Reply:
Does this mean that the other process you are referring to is in no way related to generating the response? Ultimately if you reply on the processes to generate a whole response, will the quickness of some processes matter?

Just some questions I have for someone who doesn't really know about these kind of operations. Addressing these would help make your paragraphs clearer

Reply:
>Does this mean that the other process you are referring to is in no way related to generating the response?

They are related but, some processes don't affect the user. For instance, tracking the user's activity for recommending items in the future doesn't affect the user at the current moment. So the UI can be updated before processing this.

Reply:
What you said makes sense and I can understand. Use this example and tone to make your section clearer